crypto/fipsmodule/curve25519/curve25519.c

```diff
@@ -135,16 +160,132 @@ void ED25519_keypair(uint8_t out_public_key[ED25519_PUBLIC_KEY_LEN],
   FIPS_service_indicator_update_state();
 }
 
-int ED25519_sign(uint8_t out_sig[ED25519_SIGNATURE_LEN],
-                 const uint8_t *message, size_t message_len,
+int ED25519_sign(uint8_t out_sig[ED25519_SIGNATURE_LEN], const uint8_t *message,
+                 size_t message_len,
                  const uint8_t private_key[ED25519_PRIVATE_KEY_LEN]) {
   boringssl_ensure_eddsa_self_test();
   return ED25519_sign_no_self_test(out_sig, message, message_len, private_key);
 }
 
-int ED25519_sign_no_self_test(uint8_t out_sig[ED25519_SIGNATURE_LEN],
-                              const uint8_t *message, size_t message_len,
-                              const uint8_t private_key[ED25519_PRIVATE_KEY_LEN]) {
+int ED25519_sign_no_self_test(
+    uint8_t out_sig[ED25519_SIGNATURE_LEN], const uint8_t *message,
+    size_t message_len, const uint8_t private_key[ED25519_PRIVATE_KEY_LEN]) {
+  return ed25519_sign_internal(ED25519_ALG, out_sig, message, message_len,
+                               private_key, NULL, 0);
+}
+
+int ED25519ctx_sign(uint8_t out_sig[ED25519_SIGNATURE_LEN],
+                    const uint8_t *message, size_t message_len,
+                    const uint8_t private_key[ED25519_PRIVATE_KEY_LEN],
+                    const uint8_t *context, size_t context_len) {
+  boringssl_ensure_eddsa_self_test();
+  return ED25519ctx_sign_no_self_test(out_sig, message, message_len,
+                                      private_key, context, context_len);
+}
+
+int ED25519ctx_sign_no_self_test(
+    uint8_t out_sig[ED25519_SIGNATURE_LEN], const uint8_t *message,
+    size_t message_len, const uint8_t private_key[ED25519_PRIVATE_KEY_LEN],
+    const uint8_t *context, size_t context_len) {
+  return ed25519_sign_internal(ED25519CTX_ALG, out_sig, message, message_len,
+                               private_key, context, context_len);
+}
+
+int ED25519ph_sign(uint8_t out_sig[ED25519_SIGNATURE_LEN],
+                   const uint8_t *message, size_t message_len,
+                   const uint8_t private_key[ED25519_PRIVATE_KEY_LEN],
+                   const uint8_t *context, size_t context_len) {
+  boringssl_ensure_eddsa_self_test();
```

Comment:
Is this introducing the same problem as https://github.com/aws/aws-lc/pull/2112? The first time this runs the self tests will run and increment the service indicator because it's not locked here

Reply:
Addressed, should be all good now :)